src-tests/Tests.hs

```diff
@@ -19,14 +20,15 @@ arbitraryVersion = arbitraryBoundedEnum
 arbitraryHashOptions :: Gen HashOptions
 arbitraryHashOptions =
   do p <- arbitraryWithin 1 4
-     HashOptions <$> arbitraryWithin FFI.ARGON2_MIN_TIME (min FFI.ARGON2_MAX_TIME 65536)
+     HashOptions <$> arbitraryWithin FFI.ARGON2_MIN_TIME (min FFI.ARGON2_MAX_TIME (2 ^ 13))
                  <*> arbitraryWithin (FFI.ARGON2_MIN_MEMORY*p) (FFI.ARGON2_MIN_MEMORY*p*4)  -- arbitraryWithin (max (max FFI.ARGON2_MIN_MEMORY (8 * p)) (shiftL p 3)) (min FFI.ARGON2_MAX_MEMORY 512)
                  <*> pure p
                  <*> arbitraryVariant
                  <*> arbitraryVersion
                  <*> arbitraryWithin FFI.ARGON2_MIN_OUTLEN (min FFI.ARGON2_MAX_OUTLEN 65536)
-
-arbitraryWithin lower upper = arbitrary `suchThat` (inRange (lower,upper))
```

Comment:
I'm not sure how the properties testing ever worked with this generator.
 `p <- arbitraryWithin 1 4` would blindly generate `Word32` until it a number between 1 and 4 was returned, essentially preventing progress on the suite.